server/test/unit/vitalsTest/checkButtonHeartbeatTest.js

```diff
@@ -39,225 +41,410 @@ describe('vitals.js unit tests: checkButtonHeartbeat', () => {
+        await vitals.checkButtonHeartbeat()
+        expect(helpers.logSentry).to.not.be.called
+      })
+
+      it('should not update the database', async () => {
+        await vitals.checkButtonHeartbeat()
+        expect(db.updateButtonsSentVitalsAlerts).to.not.be.called
+      })
     })
   })
 
-  describe("when client that is not sending vitals's button that is not sending vitals last seen is more recent than the threshold and an alert was sent ", () => {
+  describe('for a client that is sending vitals and has a disconnected gateway', () => {
     beforeEach(async () => {
-      this.button = buttonFactory({
-        sentVitalsAlertAt: new Date(),
-        isSendingVitals: false,
-        client: factories.clientFactory({ isSendingVitals: false }),
+      this.client = factories.clientFactory({ isSendingVitals: true })
+      sandbox.stub(db, 'getDisconnectedGatewaysWithClient').returns([gatewayFactory()])
+    })
+
+    describe('when button that is sending vitals last seen exceeds the threshold and no alerts have been sent', () => {
+      beforeEach(async () => {
+        this.button = buttonFactory({ sentVitalsAlertAt: null, isSendingVitals: true, client: this.client })
+        sandbox.stub(db, 'getButtons').returns([this.button])
+        this.buttonsVital = buttonsVitalFactory({
+          createdAt: exceededThresholdTimestamp,
+          button: this.button,
+        })
+        sandbox.stub(db, 'getRecentButtonsVitals').returns([this.buttonsVital])
       })
-      sandbox.stub(db, 'getButtons').returns([this.button])
-      this.buttonsVital = buttonsVitalFactory({
-        createdAt: currentDBDate,
-        button: this.button,
+
+      it('should log one Sentry messages', async () => {
+        await vitals.checkButtonHeartbeat()
+        expect(helpers.logSentry).to.be.calledOnce
       })
-      sandbox.stub(db, 'getRecentButtonsVitals').returns([this.buttonsVital])
-    })
 
-    it('should not send a disconnection message to Sentry', async () => {
-      await vitals.checkButtonHeartbeat()
-      expect(helpers.logSentry).to.not.be.called
-    })
+      it('should send one button disconnection message to Sentry', async () => {
+        await vitals.checkButtonHeartbeat()
+        expect(helpers.logSentry).to.be.calledWith(
+          `Disconnection: ${this.button.client.displayName} ${this.button.displayName} Button delay is ${heartbeatThreshold + 1} seconds.`,
+        )
+      })
 
-    it('should not update the database', async () => {
-      await vitals.checkButtonHeartbeat()
-      expect(db.updateButtonsSentVitalsAlerts).to.not.be.called
+      it('should not send a client message for the disconnected button', async () => {
+        expect(helpers.logSentry).to.not.be.calledWith(
+          `Button status change for: ${this.button.client.displayName}. The following buttons have been disconnected: ${this.button.displayName}.`,
+        )
+      })
     })
   })
 })
```

Comment:
There are so many test cases here; really good job being thorough! I also like how they are organized, thanks for that. :)